lib/storekit/receipt.rb

```diff
@@ -1,19 +1,42 @@
+require 'ostruct'
+
 module StoreKit
-  class Receipt
-    attr_reader :receipt_data
+  class Receipt < OpenStruct
+    TRANSFORM = {
+       'unique_identifier' => true,
+       'product_id' => true,
+       'bid' => true,
+       'unique_vendor_identifier' => true,
+       'bvrs' => true,
+       'transaction_id' => proc { |v| v.to_i },
+       'original_transaction_id' => proc { |v| v.to_i },
+       'quantity' => proc { |v| v.to_i },
+       'item_id' => proc { |v| v.to_i },
+       'web_order_line_item_id' => proc { |v| v.to_i },
+       'original_purchase_date_ms' => proc { |v| Time.at(v.to_f / 1_000) },
+       'purchase_date_ms' => proc { |v| Time.at(v.to_f / 1_000) },
+       'expires_date' => proc { |v| Time.at(v.to_f / 1_000) },
+    }
 
-    def initialize(receipt_data)
-      @receipt_data = receipt_data
-    end
+    def self.parse receipt_data
```

Comment:
I didn't implement any eager parsing of fields in the gem because:

* whether the fields are present or not in the response depends on whether the IAP is a subscription or one time purchase or consumable
* the conversions are not necessarily going to be used by the program, which means spending time making objects just for garbage collection

I think that preserving the original receipt data as it comes from Apple is very important. Wrapping the raw receipt data to provide some helper methods for type conversions on demand is useful, but not in a way that hides the original values.